source/common/router/header_formatter.h

```diff
@@ -62,5 +83,28 @@ class PlainHeaderFormatter : public HeaderFormatter {
   const bool append_;
 };
 
+/**
+ * A formatter that produces a value by concatenating the results of multiple HeaderFormatters.
+ */
+class CompoundHeaderFormatter : public HeaderFormatter {
+public:
+  CompoundHeaderFormatter(std::vector<HeaderFormatterPtr>&& formatters, bool append)
+      : formatters_(std::move(formatters)), append_(append){};
+
+  // HeaderFormatter::format
+  const std::string format(const Envoy::RequestInfo::RequestInfo& request_info) const override {
+    std::ostringstream buf;
```

Comment:
I would probably just use a raw string here and append. Any reason to use a string stream?